File: src/main/java/com/refinedmods/refinedstorage/inventory/item/validator/StorageDiskItemValidator.java
package com.refinedmods.refinedstorage.inventory.item.validator;

import com.refinedmods.refinedstorage.api.storage.disk.IStorageDiskProvider;
import net.minecraft.world.item.ItemStack;

import java.util.function.Predicate;

public class StorageDiskItemValidator implements Predicate<ItemStack> {
    @Override
    public boolean test(ItemStack stack) {
        return stack.getItem() instanceof IStorageDiskProvider && ((IStorageDiskProvider) stack.getItem()).isValid(stack);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/inventory/item/validator/StorageDiskItemValidator.kt
package com.refinedmods.refinedstorage.inventory.item.validator

import com.refinedmods.refinedstorage.api.storage.disk.IStorageDiskProvider
import net.minecraft.item.ItemStack
import java.util.function.Predicate

class StorageDiskItemValidator : Predicate<ItemStack> {
    override fun test(stack: ItemStack): Boolean {
        return stack.item is IStorageDiskProvider && (stack.item as IStorageDiskProvider).isValid(stack)
    }
}